at_client/lib/src/manager/monitor.dart

```diff
@@ -0,0 +1,259 @@
+  ///
+  /// [onResponse] function is called when a new batch of notifications are received from the server.
+  /// This cannot be null.
+  /// Example [onResponse] callback
+  /// ```
+  /// void onResponse(String notificationResponse) {
+  /// // add your notification processing logic
+  ///}
+  ///```
+  /// [onError] function is called when is some thing goes wrong with the processing.
+  /// For example this could be:
+  ///    - Unavailability of the network
+  ///    - Exception while running the code
+  /// This cannot be null.
+  /// Example [onError] callback
+  /// ```
+  /// void onError(Monitor monitor, Exception e) {
+  ///  // add your error handling logic
+  /// }
+  /// ```
+  /// After calling [onError] monitor would stop sending any more notifications. If the error is recoverable
+  /// and if [retry] is true the [Monitor] would continue and waits to recover from the error condition and not call [onError].
+  ///
+  /// For example if the app loses internet connection then [Monitor] would wait till the internet comes back and not call
+  /// [onError]
+  ///
+  /// When the [regex] is passed only those notifications matching the [regex] will be notified
+  /// When the [lastNotificationTime] is passed only those notifications AFTER the time value are notified.
+  /// This is expressed as EPOCH time milliseconds.
+  /// When [retry] is true
+  ////
+  Monitor(Function onResponse, Function onError, String atSign,
+      AtClientPreference preference, MonitorPreference monitorPreference) {
+    _onResponse = onResponse;
+    _onError = onError;
+    _preference = preference;
+    _atSign = atSign;
+    _regex = monitorPreference.regex;
+    _keepAlive = monitorPreference.keepAlive;
+    _lastNotificationTime = monitorPreference.lastNotificationTime;
+    _remoteSecondary ??= RemoteSecondary(atSign, preference);
+  }
+
+  /// Starts the monitor by establishing a new TCP/IP connection with the secondary server
+  /// If [lastNotificationTime] expressed as EPOCH milliseconds is passed, only those notifications occurred after
+  /// that time are notified.
+  /// Calling start on already started monitor would not cause any exceptions and it will have no side affects.
+  /// Calling start on monitor that is not started or erred will be started again.
+  /// Calling [Monitor#getStatus] would return the status of the [Monitor]
+  Future<void> start({int? lastNotificationTime}) async {
+    if (status == MonitorStatus.Started) {
+      // Monitor already started
+      _logger.finer('Monitor is already running');
+      return;
+    }
+    // This enables start method to be called with lastNotificationTime on the same instance of Monitor
+    if (lastNotificationTime != null) {
+      _lastNotificationTime = lastNotificationTime;
+    }
+    try {
+      await _checkConnectivity();
+      //1. Get a new outbound connection dedicated to monitor verb.
+      _monitorConnection = await _createNewConnection(
```

Comment:
I believe the code creates only one monitor for the server. It should be only one even if the start is called multiple times.

Reply:
I added atClient as a param to NotificationServiceImpl constructor..in a given app there will be only one instance of atClient for an atSign. If app switches to another atSign, a new instance of atClient will get created. 
One monitor per atSign is controlled through NotificationServiceImpl. We will not expose Monitor class to app developers